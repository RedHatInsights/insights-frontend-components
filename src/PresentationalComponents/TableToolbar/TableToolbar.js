import React, { Fragment } from 'react';
import propTypes from 'prop-types';
import { Toolbar } from '@patternfly/react-core';
import classNames from 'classnames';

import './TableToolbar.scss';

function generateResults(results) {
    if (results > 1 || results < 1) {
        return (`${results} Results`);
    } else {
        return (`${results} Result`);
    }
}

const TableToolbar = ({ isFooter, results, className, children, ...props }) => {

    const tableToolbarClasses = classNames(
        'ins-c-table__toolbar',
        { [`ins-m-footer`]: isFooter },
        className
    );

    return (
        <Fragment>
            <Toolbar className={ tableToolbarClasses } { ...props }> { children }</Toolbar>
            {
                results >= 0 &&
                <div className='ins-c-table__toolbar-results'>
                    { generateResults(results) }
                </div>
            }
        </Fragment>
    );
};

export default TableToolbar;

TableToolbar.propTypes = {
    isFooter: propTypes.bool,
    results: propTypes.number,
    children: propTypes.any,
    className: propTypes.string
};

TableToolbar.defaultProps = {
    isFooter: false
};
