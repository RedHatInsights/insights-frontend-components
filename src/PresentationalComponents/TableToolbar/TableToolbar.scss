@import '../../Utilities';

.ins-c-table__toolbar {
    background: #fff;
    border-bottom: 1px solid #ededed;
    @include rem('padding', 24px);
}