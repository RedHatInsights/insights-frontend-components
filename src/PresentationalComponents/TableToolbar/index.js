export { default as TableToolbar } from './TableToolbar.js';
