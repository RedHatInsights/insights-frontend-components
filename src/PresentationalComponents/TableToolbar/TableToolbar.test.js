import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TableToolbar from './TableToolbar';

describe('TableToolbar component', () => {
    it('should render', () => {
        const wrapper = shallow(<TableToolbar>Some</TableToolbar>)
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with results of 0', () => {
        const wrapper = shallow(<TableToolbar results={ 0 }>Some</TableToolbar>)
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with results of 1', () => {
        const wrapper = shallow(<TableToolbar results={ 1 }>Some</TableToolbar>)
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with results greater than 1', () => {
        const wrapper = shallow(<TableToolbar results={ 2 }>Some</TableToolbar>)
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})
