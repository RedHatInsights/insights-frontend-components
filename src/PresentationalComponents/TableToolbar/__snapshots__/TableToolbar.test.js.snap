// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableToolbar component should render 1`] = `
<Fragment>
  <Toolbar
    className="ins-c-table__toolbar"
  >
     
    Some
  </Toolbar>
</Fragment>
`;

exports[`TableToolbar component should render a footer 1`] = `
<Fragment>
  <Toolbar
    className="ins-c-table__toolbar ins-m-footer"
  >
     
    Footer
  </Toolbar>
</Fragment>
`;

exports[`TableToolbar component should render with results greater than 1 1`] = `
<Fragment>
  <Toolbar
    className="ins-c-table__toolbar"
  >
     
    Some
  </Toolbar>
  <div
    className="ins-c-table__toolbar-results"
  >
    2 Results
  </div>
</Fragment>
`;

exports[`TableToolbar component should render with results of 0 1`] = `
<Fragment>
  <Toolbar
    className="ins-c-table__toolbar"
  >
     
    Some
  </Toolbar>
  <div
    className="ins-c-table__toolbar-results"
  >
    0 Results
  </div>
</Fragment>
`;

exports[`TableToolbar component should render with results of 1 1`] = `
<Fragment>
  <Toolbar
    className="ins-c-table__toolbar"
  >
     
    Some
  </Toolbar>
  <div
    className="ins-c-table__toolbar-results"
  >
    1 Result
  </div>
</Fragment>
`;
